### DISCUSSION TITLE
Persistent Fleet Orders?

### DISCUSSION THREAD

**Discussion by AdmiralToroca** [endless-sky/endless-sky#10805] [General]
Apologies if this topic exists already; I _did_ try to search for it, but I am new to GitHub and it doesn't seem very intuitive to me, so I could have missed something.

Would it be possible to make orders that you give to your fleet be persistent? Meaning for instance, if you tell them to gather around the flagship, they will do so and will **STAY** around the flagship until given a new order. If this is already possible, I have not been able to find a setting or command for it.

I found the game on GOG less than a month ago, and have been enjoying it quite a lot even though I know I've barely scratched the surface of what you can do, but this one issue is possibly the single-most annoying problem to me. I've upgraded to stronger ships now, but prior to that I lost a couple when doing my best to irritate the Korath or when encountering large pirate fleets. The reason being, even though I had ordered my ships to gather around the flagship, at some point they went flying off on their own after individual targets or to grab flotsam, and before I realized it and called them back, they got swarmed and blown to bits by enemies that the combined fleet could have withstood.

And even though I have tougher ships now, it still drives me up the wall that they'll take off on their own whenever they feel like it even after having been ordered to gather around the flagship.

My suggestion would be to make orders given to your fleet or individual ships be persistent until superseded by a new order or until a new "Clear existing orders" command were issued to send them back to default behavior. I would use this most frequently with the Gather command to concentrate my firepower and protect my freighters, but I can foresee uses for it with some of the other fleet commands as well.

**Comment by esksp** [comment]
Orders and their persistence is a pet peeve for me too, unless I missed something obvious (it has happened before...).
My biggest issue is with the "autopilot" or "targeting" command: Apparently it stays on or switches off with no discernible reason, so sometimes I'm wasting time shooting empty space, and other times my ship is always turning towards some target even if I don't want it to (like, when my ship gets too damaged and I'd like to rush straight out of there - fast!).


> I found the game on GOG

So did I. I'd suggest you download the latest version though, from here (https://github.com/endless-sky/endless-sky/releases/tag/v0.10.10). Especially since the 0.10.8 GOG version (Linux) I downloaded September had a bug (which has been fixed since though).


> make orders given to your fleet or individual ships be persistent until superseded by a new order

...and have some **visual cue which orders are currently active**. A single letter (like "G" for "gather") would be enough to allow you to know what's currently going on.

**Reply by AdmiralToroca** [reply]
I got v0.10.10 a few weeks ago now, but thanks for pointing it out. :) I _did_ notice the other day while browsing their black friday specials that GOG has updated to v0.10.10 as well now, but it was indeed still v0.10.08 when I first found it. I'm not positive, but I think I actually first found it between the time 10.10 was released on GitHub and when GOG updated their version to match. So I had initially downloaded 10.08, but found 10.10 literally within a day or two as I started looking for more info about the game.

Figuring out how to upgrade my version was a whole other nightmare though, being unfamiliar with GitHub and not knowing how long it would take GOG to update theirs. I couldn't find any answer as to whether I could install over the top of the old version or needed a clean install. I ended up playing it safe and putting 10.10 in a new directory alongside 10.08. Which reminds me, I don't think I ever actually _removed_ 10.08.  :D

As for the actual topic of this thread, your idea about a visual representation seems good; probably somewhere on the HUD display of the ships rather than in the main screen area? Similar to my original suggestion about making orders more persistent, I've started wondering whether it would be possible to have multiple persistent orders simultaneously, but ranked by priority.

As an example: say you've got a few ships and you're running around mining. You could issue the Gather command to keep your other ships (and their cargo bays) nearby, but also issue the Collect Flotsam order as a secondary one so that when you blow up an asteroid they would automatically grab the minerals but then immediately return to your flagship as soon as everything was collected, without having to be told again to Gather. Then you blow up the next asteroid and they repeat the process, without you having to manually tell them "Go get it it!" and "Okay, come back now!"

It could work almost the same with battles; ship blows up, they grab whatever it dropped, and then come right back. I'd imagine for a system like this to work and not create its own new annoyances, there'd have to be some sort of range limitation on the secondary order to prevent your fleet from running halfway across the map to grab stuff that you weren't even aware was there, like flotsam from Pirates that were destroyed by Republic ships somewhere far from your flagship. Maybe something like twice the range of the better beam weapons, or hey, even allow us to set the range limit ourselves? That'd be cool.

All of that said, however, I would be perfectly happy with the simpler, original version of my suggestion, where they'd just stick to the flagship until I tell them they can go do something else.  ;)

**Reply by esksp** [reply]
> whether I could install over the top of the old version

This, or just install it in a new directory and delete the old one. After all the Windows installer you download is a simple ZIP file, there are no registry shenanigans going on... GOG does add their own branding stuff in the installation, but those files don't affect the game itself and aren't important to have.

> somewhere on the HUD display of the ships 

Where ever it makes sense: For actions affecting your flagship (like the aforementioned auto-targeting), it would make sense to have it on the main screen, maybe somewhere near the secondary weapons list? For actions affecting the escorts, it would make sense to have it near the escorts list.

> issue the Gather command [...] but also issue the Collect Flotsam order

Or, they could just implement **a "collect flotsam" action**, where escorts do exactly what you said: They swarm out collecting any usable morsels, and then _automatically_ come back to land on your flagship. And stay there. It would also minimize the time your vulnerable escorts are out there, especially with a battle raging nearby.

> prevent your fleet from running halfway across the map to grab stuff

Yes, **escorts definitely need to have a range limitation** ("**only go so far from your carrier**"), if only for credibility's sake.
Also because I doubt there are many players out there enjoying the "herding cats" feeling when in the midst of a difficult situation all your escorts decide to do something totally unimportant and terribly stupid. :-(

**Comment by 9Pigs** [comment]
Hold position sticks **only** until you land. I found that out when I wanted to do a single ship sneak-in job some distance away. I had to refuel on the way and was surprised as my fleet joined me.  Yea, I now know bout "parking" -- but that is a deep dive into the User Interface dance.

  And yea, I agree that the focus fire should have better default when the target is done.  It seems the default is melee mode, instead of  organized battle plan mode (like gather to the mother ship).

**Comment by CrispySummers** [comment]
I agree with that!

I hope I'm not being rude, but if you are a coder, you might be able to implement this quite easily by diving into the ES Codebase and modifiying it to save fleet orders to memory (until given new orders) and to your savefile so they'll persist across game sessions. The Codebase is well-organized and clean, as far as I've seen :)
